libraries/AP_GPS/AP_GPS.h

```diff
@@ -144,8 +144,16 @@ class AP_GPS
         uint32_t last_gps_time_ms;          ///< the system time we got the last GPS timestamp, milliseconds
 
         // all the following fields must only all be filled by RTK capable backend drivers
+        uint32_t rtk_time_week_ms;         ///< GPS Time of Week of last baseline in milliseconds
+        uint16_t rtk_week_number;          ///< GPS Week Number of last baseline
         uint32_t rtk_age_ms;               ///< GPS age of last baseline correction in milliseconds  (0 when no corrections, 0xFFFFFFFF indicates overflow)
         uint8_t  rtk_num_sats;             ///< Current number of satellites used for RTK calculation
+        uint8_t  rtk_baseline_coords_type; ///< Coordinate system of baseline. 0 == ECEF, 1 == NED
+        int32_t  rtk_baseline_x_mm;        ///< Current baseline in ECEF x or NED north component in mm
+        int32_t  rtk_baseline_y_mm;        ///< Current baseline in ECEF y or NED east component in mm
+        int32_t  rtk_baseline_z_mm;        ///< Current baseline in ECEF z or NED down component in mm
+        uint32_t rtk_accuracy;             ///< Current estimate of 3D baseline accuracy (receiver dependent, typical 0 to 9999)
```

Comment:
This remains an absolutely frustratingly meaningless field. An accuracy estimate must have a unit on it, and as such this is absolutely GPS device specific behavior. I get that this was addded by emlid, for emlid, but I really hate it, because it has no meaningful unit associated with it, and there is no way to apply any other driver to it sanely.

With all that being said, I guess we can report it because there is a MAVLink field for it? I just really hate that its useless information. (Really IAR hypotheses is to the best of my ability to tell also useless information in flight and only needed for initial setup).

Reply:
IAR hypotheses is useful to see from an end user to help determine if the RTK is close to getting a Fixed Fixed (IAR of 3) when it is in Floating Fix mode.

Reply:
The existing Mavlink message is here: http://mavlink.org/messages/common#GPS_RTK
This patch only address some of those fields because:

- Emlid refused to provide the information in some of the missing fields
- Some reviews I got said that some of the fields are meaningless and unimportant
- I was lazy to implement some of them

So this is a compromise :)